platform/default/src/mbgl/storage/offline_database.cpp

```diff
@@ -1278,6 +1266,19 @@ bool OfflineDatabase::exceedsOfflineMapboxTileCountLimit(const Resource& resourc
         && offlineMapboxTileCountLimitExceeded();
 }
 
+void OfflineDatabase::markUsedResources(int64_t regionID, const std::list<Resource>& resources) try {
+    if (!db) {
```

Comment:
db should be initialized in constructor, assert(db).

Reply:
There are code paths resetting db, it's taken from other public methods. I think the contract is that DB must initialize lazily.